.github/workflows/databricks-cicd.yml

```diff
@@ -0,0 +1,142 @@
+name: Databricks_CI_CD
+
+on:
+  push:
+    paths:
+      - "samples/databricks-cicd/**"
+      - "!samples/databricks-cicd/README.md"
+      - "!samples/databricks-cicd/.env.sample"
+  pull_request:
+    branches: [main]
+    paths:
+      - "samples/databricks-cicd/**"
+      - "!samples/databricks-cicd/README.md"
+      - "!samples/databricks-cicd/.env.sample"
+
+  # Allows you to run this workflow manually from the Actions tab
+  # workflow_dispatch:
+
+env:
+  SAMPLE_DIRECTORY: samples/databricks-cicd
+
+jobs:
+  build:
+    environment: Databricks_Azure_Test
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ${{ env.SAMPLE_DIRECTORY }}
+    env:
+      DATABRICKS_HOST: ${{ secrets.DATABRICKS_HOST }}
+      DATABRICKS_TOKEN: ${{ secrets.DATABRICKS_TOKEN }}
+      DATABRICKS_CLUSTERID: ${{ secrets.DATABRICKS_CLUSTERID }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup Python
+        uses: actions/setup-python@v2.2.2
+        with:
+          python-version: 3.7
+
+      - name: Setup Miniconda
+        uses: conda-incubator/setup-miniconda@v2.1.1
+        with:
+          environment-file: ${{ env.SAMPLE_DIRECTORY }}/environment.yml
+          activate-environment: databricks_cicd
+          python-version: 3.7
+
+      - name: Lint
+        shell: bash -l {0}
+        run: |
+          flake8
+          black .
```

Comment:
I also found out about `black` lately. Wouldn't it make more sense to run black in a pre commit hook, that the code format is guaranteed on commit?

In this case we would check if linting rules are ok and afterwards format the files but this won't have any effect on the running code and doesn't reflect back to the repo. It could even lead to confusion if e.g. code lines are changed and in the next steps you get an error at line `x` wit code `z` but you won't find that line with that code in your repo.

I also saw it used the other way round, executing first black in the pipeline and then run the linter. But same here you have unformatted code in the repo and make it nice before the linter runs. Feels like a bit like cheating ;)

Honestly maybe I didn't understand what `black` does exactly. I suggested it will format the code. Hope you can help me out here.

Reply:
Good call. At least in the CI pipeline, it should be checking not formatting: `black --check .` 

As for pre-commit config, also agree. Although I've set VSCode to do formatting on save and linting. Pre-commit would be an additional level of protection. It still depends on each developer setting it up. The check in the CI pipeline makes sure it's consistent across the dev team I think.

Reply:
Oh right, good point to check it in addition to align it across the team. I guess there are also some possibilities in running post create scripts on the devcontainer side to set up pre commits and other tooling. The need depends a bit if you are working with a more or less stable crew. For open source contribution I guess it would be great to automate credential scanning and additional pre commit setup into the devcontainer as well.